annotationsx/middleware.py

```diff
@@ -134,4 +134,23 @@ def process_response(self, request, response):
                                 path=settings.SESSION_COOKIE_PATH,
                                 secure=settings.SESSION_COOKIE_SECURE or None,
                                 httponly=settings.SESSION_COOKIE_HTTPONLY or None)
-        return response
\ No newline at end of file
+        return response
+
+
+class CookielessSessionMiddleware(object):
+    def __init__(self):
+        engine = import_module(settings.SESSION_ENGINE)
+        self.SessionStore = engine.SessionStore
+        print "STarting engine"
+
+    def process_request(self, request):
+        session_key = request.COOKIES.get(settings.SESSION_COOKIE_NAME, request.GET.get('utm_source'))
```

Comment:
My understanding is that it should have a value regardless of which middleware is selected, since it appears to be defined in the  [global settings](https://github.com/django/django/blob/master/django/conf/global_settings.py), and not the middleware. It looks like unless you override it, it will always have the default value of **sessionid**.